core/src/banking_stage/transaction_scheduler/transaction_state_container.rs

```diff
@@ -81,27 +79,35 @@ pub(crate) trait StateContainer<Tx: TransactionWithMeta> {
             .expect("transaction must exist");
         let priority_id = TransactionPriorityId::new(transaction_state.priority(), transaction_id);
         transaction_state.transition_to_unprocessed(transaction_ttl);
-        self.push_id_into_queue(priority_id);
+        self.push_ids_into_queue(std::iter::once(priority_id));
     }
 
-    /// Pushes a transaction id into the priority queue. If the queue is full, the lowest priority
-    /// transaction will be dropped (removed from the queue and map).
-    /// Returns `true` if a packet was dropped due to capacity limits.
-    fn push_id_into_queue(&mut self, priority_id: TransactionPriorityId) -> bool;
+    /// Pushes transaction ids into the priority queue. If the queue if full,
+    /// the lowest priority transactions will be dropped (removed from the
+    /// queue and map) **after** all ids have been pushed.
+    /// To avoid allocating, the caller should not push more than
+    /// [`EXTRA_CAPACITY`] ids in a call.
+    /// Returns the number of dropped transactions.
```

Comment:
Is it enough to enforce that you cannot push any ids that aren't already in the tx container slab? As long as the tx container never goes over `self.capacity + EXTRA_CAPACITY` we can never get into a situation where we over allocate here I think?

Reply:
We shouldn't be pushing anything to priority queue that's not in the map already.

Reply:
Cool, then we don't really need this comment right? We can push however many ids that we want (even more than `EXTRA_CAPACITY` ids) in a call because we know that the resulting length will never be more than the tx container map size.

I was just concerned that if `EXTRA_CAPACITY` was truly a hard limit, we're not enforcing it anywhere and if batch size was ever higher than `EXTRA_CAPACITY`, we could start pushing more than `EXTRA_CAPACITY` into the priority queue when a full batch of transactions fails and needs to be requeued.